src/ApplicationInsights.Kubernetes/ContainerIdProviders/MountInfoContainerIdProvider.cs

```diff
@@ -0,0 +1,65 @@
+#nullable enable
+
+using System;
+using System.IO;
+using System.Text.RegularExpressions;
+using Microsoft.ApplicationInsights.Kubernetes.Debugging;
+
+namespace Microsoft.ApplicationInsights.Kubernetes.ContainerIdProviders;
+
+internal class MountInfoContainerIdProvider : IContainerIdProvider
+{
+    private const string InfoFilePath = "/proc/self/mountinfo";
+    private const string MatchPattern = @"\/docker\/containers\/(.*?)\/";
+    private static readonly Regex MatchRegex = new Regex(MatchPattern, RegexOptions.CultureInvariant | RegexOptions.Multiline, TimeSpan.FromSeconds(1));
```

Comment:
Why `Multiline`? In the code below you are reading the file one line at a time...

Reply:
Fixed.